File: retrofit-response-type-keeper/src/test/resources/all-http-methods/Service.java
package test;

import retrofit2.*;
import retrofit2.http.*;

class DeleteUser {}
class GetUser {}
class HeadUser {}
class HttpUser {}
class OptionsUser {}
class PatchUser {}
class PostUser {}
class PutUser {}

interface Service {
  @DELETE("/") Call<DeleteUser> delete();
  @GET("/") Call<GetUser> get();
  @HEAD("/") Call<HeadUser> head();
  @HTTP(method = "CUSTOM", path = "/") Call<HttpUser> http();
  @OPTIONS("/") Call<OptionsUser> options();
  @PATCH("/") Call<PatchUser> patch();
  @POST("/") Call<PostUser> post();
  @PUT("/") Call<PutUser> put();
}

File: retrofit-response-type-keeper/src/test/resources/all-http-methods/Service.kt
package test

import retrofit2.*
import retrofit2.http.*

internal class DeleteUser
internal class GetUser
internal class HeadUser
internal class HttpUser
internal class OptionsUser
internal class PatchUser
internal class PostUser
internal class PutUser

internal interface Service {
  @DELETE("/")
  fun delete(): Call<DeleteUser>

  @GET("/")
  fun get(): Call<GetUser>

  @HEAD("/")
  fun head(): Call<HeadUser>

  @retrofit2.http.HTTP(method = "CUSTOM", path = "/")
  fun http(): Call<HttpUser>

  @OPTIONS("/")
  fun options(): Call<OptionsUser>

  @PATCH("/")
  fun patch(): Call<PatchUser>

  @POST("/")
  fun post(): Call<PostUser>

  @PUT("/")
  fun put(): Call<PutUser>
}
